main.go

```diff
@@ -140,178 +194,327 @@ func main() {
+
+	log.SetEnableDebugLog(configs.VerboseLog == "yes")
 
 	archiveExt := filepath.Ext(configs.ArchivePath)
 	archiveName := filepath.Base(configs.ArchivePath)
 	archiveName = strings.TrimSuffix(archiveName, archiveExt)
-
 	appPath := filepath.Join(configs.DeployDir, archiveName+".app")
 	pkgPath := filepath.Join(configs.DeployDir, archiveName+".pkg")
 	exportOptionsPath := filepath.Join(configs.DeployDir, "export_options.plist")
-
 	ideDistributionLogsZipPath := filepath.Join(configs.DeployDir, "xcodebuild.xcdistributionlogs.zip")
 
+	xcodebuildVersion, err := utility.GetXcodeVersion()
+	if err != nil {
+		fail("Failed to determin xcode version, error: %s", err)
+	}
+	log.Printf("- xcodebuildVersion: %s (%s)", xcodebuildVersion.Version, xcodebuildVersion.BuildVersion)
+
+	customExportOptionsPlistContent := strings.TrimSpace(configs.CustomExportOptionsPlistContent)
+	if customExportOptionsPlistContent != configs.CustomExportOptionsPlistContent {
+		fmt.Println()
+		log.Warnf("CustomExportOptionsPlistContent is stripped to remove spaces and new lines:")
+		log.Printf(customExportOptionsPlistContent)
+	}
+
 	envsToUnset := []string{"GEM_HOME", "GEM_PATH", "RUBYLIB", "RUBYOPT", "BUNDLE_BIN_PATH", "_ORIGINAL_GEM_PATH", "BUNDLE_GEMFILE"}
 	for _, key := range envsToUnset {
 		if err := os.Unsetenv(key); err != nil {
 			fail("Failed to unset (%s), error: %s", key, err)
 		}
 	}
 
-	if configs.UseLegacyExport == "yes" {
-		log.Info("Using legacy export method...")
+	archive, err := xcarchive.NewMacosArchive(configs.ArchivePath)
+	if err != nil {
+		fail("Failed to parse archive, error: %s", err)
+	}
 
-		provisioningProfileName := ""
-		if configs.LegacyExportProvisioningProfileName != "" {
-			log.Detail("Using provisioning profile: %s", configs.LegacyExportProvisioningProfileName)
-			provisioningProfileName = configs.LegacyExportProvisioningProfileName
-		} else {
-			log.Detail("No provisining profile specified, let xcodebuild to grab one...")
-		}
+	// do a simple export if method set to none
+	{
+		if configs.ExportMethod == "none" {
+			log.Infof("Exporting app without re-sign...")
+
+			if err := utils.ExportAppFromArchive(configs.ArchivePath, configs.DeployDir, bitriseAppPathEnvKey); err != nil {
+				fail("Failed to archive app, error: %s", err)
+			}
 
-		exportingApp := true
-		if configs.LegacyExportOutputFormat == "pkg" {
-			exportingApp = false
+			log.Donef("The app path is now available in the Environment Variable: %s", bitriseAppPathEnvKey)
+			return
 		}
+	}
 
-		legacyExportCmd := xcodebuild.NewLegacyExportCommand()
-		legacyExportCmd.SetExportFormat(configs.LegacyExportOutputFormat)
+	exportMethod, err := exportoptions.ParseMethod(configs.ExportMethod)
+	if err != nil {
+		fail("Failed to parse export options, error: %s", err)
+	}
 
-		if exportingApp {
-			legacyExportCmd.SetExportPath(appPath)
-		} else {
-			legacyExportCmd.SetExportPath(pkgPath)
-		}
+	// legacy export
+	{
+		if configs.UseLegacyExport == "yes" {
```

Comment:
fail if legacy options are set and xcode major version >= 9 like: https://github.com/bitrise-io/steps-xcode-archive/pull/95/files#diff-7ddfb3e035b42cd70649cc33393fe32cR472